src/main/java/appeng/client/gui/implementations/GuiPatternTerm.java

```diff
@@ -78,34 +80,43 @@ protected void actionPerformed( final GuiButton btn )
 			{
 				NetworkHandler.instance.sendToServer( new PacketValueConfig( "PatternTerminal.Clear", "1" ) );
 			}
+
+			if( this.substitutionsEnabledBtn == btn || this.substitutionsDisabledBtn == btn )
+			{
+				NetworkHandler.instance.sendToServer( new PacketValueConfig( "PatternTerminal.Substitute", this.substitutionsEnabledBtn == btn ? "0" : "1" ) );
```

Comment:
Might be even an idea to refactor `PacketValueConfig` to also accept an `enum` instead of just a random string. Serialization probably still has to fall back to `.ordinal()` or similar.